pilot/cmd/pilot-agent/status/server.go

```diff
@@ -546,12 +547,18 @@ func (s *Server) handleAppProbe(w http.ResponseWriter, req *http.Request) {
 		appReq.Header[name] = newValues
 	}
 
-	for _, h := range prober.HTTPGet.HTTPHeaders {
-		if h.Name == "Host" || h.Name == ":authority" {
-			// Probe has specific host header override; honor it
-			appReq.Host = h.Value
-		} else {
-			appReq.Header.Set(h.Name, h.Value)
+	if headers := prober.HTTPGet.HTTPHeaders; len(headers) != 0 {
+		for _, h := range headers {
+			delete(appReq.Header, h.Name)
```

Comment:
What if incoming request has 
```
a: [b,c]
```

and HTTPHeaders has
```
a: [b]
```

?

I am not even sure why we have both forwarding headers *and* reading HTTPHeaders? Shouldn't we do one or the other?

Reply:
I thought we forwarded everything except for overwriting probe headers

Reply:
>  its for user agent header. So we should keep both

https://github.com/istio/istio/pull/23504/files#r433529535

Reply:
Avoid recurrence of the problem solved by the PR #28537